lib/bitcoin/lock/LockResolver.ts

```diff
@@ -85,9 +85,11 @@ export default class LockResolver {
     const lockStartBlock = await this.calculateLockStartingBlock(lockTransaction);
 
     // (C). verify that the lock duration is valid
-    if (!this.isLockDurationValid(lockStartBlock, scriptVerifyResult.unlockAtBlock!)) {
+    const unlockAtBlock = lockStartBlock + scriptVerifyResult.lockDurationInBlocks!;
+
+    if (!this.isLockDurationValid(lockStartBlock, unlockAtBlock)) {
       throw new SidetreeError(ErrorCode.LockResolverDurationIsInvalid,
-                              `Lock start block: ${lockStartBlock}. Unlock block: ${scriptVerifyResult.unlockAtBlock}`);
+                              `Lock start block: ${lockStartBlock}. Unlock block: ${unlockAtBlock}`);

```

Comment:
And then this could be `valid duration is x, but got y duration`

Reply:
Good idea .. let me see how can I improve this log without having to recalculate what's inside the function.